Pixel Soup

Franccisco opened the pull request.

Commit:
Integrate pipenv to the workflow

Commit:
Add README.md

Commit:
Pre commit integration (#1)

* Add flake8 into pre-commit configurations

* Add black into pre-commit configurations

* Add black badge in README.md

* Add pre-commit badge in README.md

Commit:
Disable pre-releases from pipfile.lock
To make the black package work with pipenv, allow_prereleases = true was needed. As it is used directly in the pre-commit hook (and not in the pipenv), there is no need to keep allow_prereleases.

Commit:
Add the skeleton of the game
This file contains the entrypoint of the game. Currently it is only a skeleton, with minimal functions to run the game.

Commit:
Add the "start" script to pipenv
With this, the game can be started using pipenv run start

Commit:
Merge pull request #2 from Franccisco/game-skeleton

Add game skeleton

Commit:
Move constants to gameconstants.py
With this change, the constants will be in a different file, specific to its purpose. One of the reasons behind this change is that it is clearer to say from `gameconstants import SCREEN_WIDTH` than `from main import SCREEN_WIDTH`.

Commit:
Add Continuous Integration (#4)

Use flake8 and black to check code style

Commit:
Revert "Add Continuous Integration (#4)"

This reverts commit f742d0ac38309e74af111a51edc76bb538fbe117.

Commit:
Add MainView

Commit:
Add MainMenuView

Commit:
Add GameView

Commit:
Add PauseView

Commit:
Add LoadGameView

Commit:
Add SettingsView

davidoluwafemi committed:
Merge pull request #6 from Franccisco/views

Add Views

davidoluwafemi committed:
Add Networking support (#13)

* Restructuring project, adding backend code and net_interface to frontend

* A little documentation and typo fix

* Update "start" script in Pipfile
The internal structure of the project has changed, therefore the start script is modified to point to the new location of main.py.

Co-authored-by: Franccisco <35090042+Franccisco@users.noreply.github.com>

Commit:
Add RoomView
Instead of giving the user the option of starting a new game or loading an existing one, the user will now be able to create or join a multiplayer room. That said:

- LoadGameView is no longer needed.
- NewGameButton is renamed to NewRoomButton
- LoadGameButton is renamed to JoinRoomButton

Commit:
Add Networking connection to the roomview
It is needed to:
- Define the SERVER and PORT environment variables in a .env file located at Pixel Soup directory
- Run the backend.py file located in src/backend/backend.py.

Commit:
Add a DialogueBox to the MainMenu Viev
This DialogueBox will appear when the user clicks on the New Room or Join Room buttons, so the user can establish a connection to the server.

davidoluwafemi committed:
Merge pull request #16 from Franccisco/room-view

Room view

davidoluwafemi committed:
Integrate networking with the frontend (#17)

* bullet implementation and gravity

* new game concept

* optimising net_interface

* backend fix

* file renaming

* adding ntworking interface to test directory

* restructure and backend fix

* Put the recent changes in the appropiate files and folders

* Remove unused import

* Remove unused view

* Add PORT to .env

Co-authored-by: Franccisco <35090042+Franccisco@users.noreply.github.com>

Commit:
Make a functional RoomView
With the usage of a Thread, RoomView is not being blocked, and once the team is full, all the players are redirected to the GameView.

Commit:
Make each player use a unique character
The last packet sent before the beggining of the game contains a list of the usernames in the order they joined the room, so that's used to assign each player a unique character.

Commit:
Check correctly if viable data is available to be sent
In the backend file, the modified if statement was checking in a way that in case pipe[data[1]] doesn't exist, an exception is raised.

Commit:
Remove unused StartGame button

Commit:
Merge pull request #18 from Franccisco/functional-room-view

Functional room view

Commit:
Schedule the wall addition
Instead of calling `stream` on every update, it is scheduled so the time the method is called can be controlled.

davidoluwafemi committed:
Merge pull request #19 from Franccisco/game-logic

Game logic

davidoluwafemi committed:
establishing connection between waiting room and gameroom over networking

davidoluwafemi committed:
implementing jumping, falling sound and background pictures

Commit:
Fix linting errors due to merge conflicts

Commit:
Merge pull request #20 from Franccisco/project_fix

Backend restructuring

davidoluwafemi committed:
fixes

davidoluwafemi committed:
smart platform generator

Commit:
Merge pull request #21 from Franccisco/final_fixes

Final fixes

davidoluwafemi committed:
platformer speed

Commit:
Drop SettingsView
In the current game structure, this view is not being used.

Commit:
Merge pull request #22 from Franccisco/drop-settings-view

Drop SettingsView

davidoluwafemi committed:
server identity recognition

Commit:
Merge pull request #23 from Franccisco/final_fixes

Add server identity recognition

Commit:
gameview fix (#24)

Co-authored-by: Inventor <1inventordavid@gmail.com>

Commit:
Add a Dialogue Box with the key bindings (#25)

Commit:
Drop initial joystick support

Commit:
Merge pull request #26 from Franccisco/drop-initial-joystick-support

Drop initial joystick support

Commit:
README (#14)

* Add an initial sketch in README

* Add instructions to install the game

* Add Game Title to README.md

* Add instructions to run the game

* Add LICENSE
This file matches the license of the parent repository, but adding the contributors of this game.

* Add LICENSE to README.md

* Add a brief description of the game

* Fix typo in License section

* Add a brief description of Usage

* Add main_view.png

* Update key binding to start the game

Commit:
Merge pull request #5 from Franccisco/revert-4-github-actions

Revert "Add Continuous Integration"

kwzrd commented:
Hi @Franccisco, we haven't merged your PR yet because it adds a file outside of your project folder named `.pre-commit-config.yaml`.

We ask that you please remove this file in an extra commit - due to the nature of the file it shouldn't have any effect on your project.

Please make sure you do not make any other changes in this commit. Once this is done, we will merge your project. Thanks!

Commit:
Delete .pre-commit-config.yaml

According to the jam rules, this file should be only in the project's directory (Pixel Soup in this case).

ghost commented:
@kwzrd I'm sorry about that. I just deleted that file.

kwzrd commented:
@Franccisco No worries, and thanks for fixing it up!
